taglib/toolkit/tutils.h

```diff
@@ -203,11 +203,12 @@ namespace TagLib
       /*!
        * Returns the byte order of the system.
        */
-      inline ByteOrder systemByteOrder()
+      inline constexpr ByteOrder systemByteOrder()
       {
-        union {
+        union IntCharUnion {
           int  i;
           char c;
+          constexpr IntCharUnion() : i{} {}
```

Comment:
is this even needed? i can be default initialized AFAIK.

Reply:
If I use

```
union {
  int  i;
  char c;
} u = { 0 };
```

GCC accepts it, but Clang complains

```
taglib/toolkit/tutils.h:214:12: note: read of member 'c' of union with active member 'i' is not allowed in a constant expression
        if(u.c == 1)
```

Reply:
I meant int I{};

Reply:
```
union {
  int  i {};
  char c;
} u;
```

Clang accepts it, but GCC complains

```
error: uninitialized variable ‘u’ in ‘constexpr’ function
```

Reply:
What a disaster. OK